lib/whitehall_importer.rb

```diff
@@ -23,9 +25,26 @@ def self.create_migration(organisation_content_id, document_type)
     end
   end
 
-  def self.import_and_sync(whitehall_document)
-    whitehall_import = WhitehallMigration::DocumentImport.create!(
-      whitehall_document_id: whitehall_document["id"],
+  def self.queue_migration_jobs(whitehall_migration)
+    documents = WhitehallMigration::DocumentImport.where(whitehall_migration_id: whitehall_migration.id)
+    documents.each do |document|
+      WhitehallDocumentImportJob.perform_later(document)
+    end
+  end
+
+  def self.mark_migration_completed(whitehall_migration_document_import)
+    running_states = %w(pending importing imported syncing)
+    whitehall_migration = WhitehallMigration.find(whitehall_migration_document_import.whitehall_migration_id)
+    if WhitehallMigration::DocumentImport.where(whitehall_migration_id: whitehall_migration.id, state: running_states).empty?
```

Comment:
These 3 lines seem a bit funky and I think they could be wrote more declaratively with ActiveRecord associations.

We could put a scope on DocumentImport to determine which ones are finished with something like:  `scope :in_progress, -> { pending.or(importing).or(imported).or(syncing) }` and then on the whitehall_migration model ask `whitehall_migration.documents.in_progress.any?`

This will mean setting up the association in ActiveRecord but we should do that anyway since these are associations.

Reply:
This is definitely much cleaner and has reduced the length/complexity of this code.